transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,117 @@
+
+== Issue Metadata
+
+=== Issue
+
+* [WFLY-17742](https://issues.jboss.org/browse/WFLY-17742)
+
+=== Dev Contacts
+
+* mailto:{jfinelli@redhat.com}[{Manuel Finelli}]
+
+=== QE Contacts
+
+* mailto:{istraka@redhat.com}[{Ivan Straka}]
+
+=== Testing By
+
+* [ ] Engineering
+* [X] QE
+
+=== Affected Projects or Components
+
+* [wildfly-core](https://github.com/wildfly/wildfly-core)
+* [wildfly](https://github.com/wildfly/wildfly)
+* [narayana](https://github.com/jbosstm/narayana)
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* This proposal is about importing into WildFly the transactional functionalities of its Kubernetes Operator. It does not include any further improvements
+* The Management Model of WildFly will not be modified
+* The Transactions subsystem needs to become aware of in-doubt and heuristic transactions and delay its suspension as long as all transactions are completed. This will guarantee data integrity in all scenarios, including situations when a negative timeout (i.e. indefinite waiting) is employed (e.g. in cloud environments)
+* The administrator needs to receive notifications when the Transactions subsystem is causing delays in the graceful shutdown of WildFly, achieved through integration with the existing logging mechanism. This is especially true when a negative timeout is used (i.e. indefinite graceful shutdown). Moreover, internal details like 'ServerActivity' do not need to be prominent. Part of working out such a notification would be how to consistently describe a ServerActivity to the administrator in an understandable manner without exposing internal technical implementation details
+
+=== Nice-to-Have Requirements
+
+* Introducing an asynchronous graceful shutdown would enable all ServerActivity implementations to receive a pre-suspend/suspend signal concurrently. In this way, the timeout will have the same duration for all ServerActivity implementations.
+* [The sequence to suspend](https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/suspend/SuspendController.java#L57-L99) ServerActivity implementations during WildFly’s graceful shutdown should be Last Input First Output (LIFO), i.e. the last ServerActivity implementation that was registered during startup should be the first one to get suspended.
```

Comment:
Remark, source references become outdated if the source changes. Can you reference a tag instead of the main branch, [like this](https://github.com/wildfly/wildfly-core/blob/22.0.1.Final/server/src/main/java/org/jboss/as/server/suspend/SuspendController.java#L84), for example.